pyscf/adc/uadc.py

```diff
@@ -1829,89 +2111,91 @@ def ea_adc_diag(adc,M_ab=None,eris=None):
+#            
+#            temp = np.zeros((nocc_a, nvir_a, nvir_b))
+#            temp[:] += np.diag(eris_vVvV).reshape(nvir_a,nvir_b)
+#            temp = np.ascontiguousarray(temp.transpose(0,2,1))
+#            diag[s_aba:f_aba] += temp.reshape(-1)
+#                
+#            eris_ovov_p = np.ascontiguousarray(eris_oovv.transpose(0,2,1,3))
+#            eris_ovov_p -= np.ascontiguousarray(eris_ovvo.transpose(0,2,3,1))
+#            eris_ovov_p = eris_ovov_p.reshape(nocc_a*nvir_a, nocc_a*nvir_a)
+#  
+#            temp = np.zeros((eris_ovov_p.shape[0],nvir_a))
+#            temp.T[:] += np.diagonal(eris_ovov_p)
+#            temp = temp.reshape(nocc_a, nvir_a, nvir_a)
+#            diag[s_aaa:f_aaa] += -temp[:,ab_ind_a[0],ab_ind_a[1]].reshape(-1)
+#
+#            temp = np.ascontiguousarray(temp.transpose(0,2,1))
+#            diag[s_aaa:f_aaa] += -temp[:,ab_ind_a[0],ab_ind_a[1]].reshape(-1)
+#
+#            eris_OVOV_p = np.ascontiguousarray(eris_OOVV.transpose(0,2,1,3))
+#            eris_OVOV_p -= np.ascontiguousarray(eris_OVVO.transpose(0,2,3,1))
+#            eris_OVOV_p = eris_OVOV_p.reshape(nocc_b*nvir_b, nocc_b*nvir_b)
+#  
+#            temp = np.zeros((eris_OVOV_p.shape[0],nvir_b))
+#            temp.T[:] += np.diagonal(eris_OVOV_p)
+#            temp = temp.reshape(nocc_b, nvir_b, nvir_b)
+#            diag[s_bbb:f_bbb] += -temp[:,ab_ind_b[0],ab_ind_b[1]].reshape(-1)
+#
+#            temp = np.ascontiguousarray(temp.transpose(0,2,1))
+#            diag[s_bbb:f_bbb] += -temp[:,ab_ind_b[0],ab_ind_b[1]].reshape(-1)
+#
+#            temp = np.zeros((nvir_a, nocc_b, nvir_b))
+#            temp[:] += np.diagonal(eris_OVOV_p).reshape(nocc_b, nvir_b)
+#            temp = np.ascontiguousarray(temp.transpose(1,0,2))
+#            diag[s_bab:f_bab] += -temp.reshape(-1)
+#
+#            temp = np.zeros((nvir_b, nocc_a, nvir_a))
+#            temp[:] += np.diagonal(eris_ovov_p).reshape(nocc_a, nvir_a)
+#            temp = np.ascontiguousarray(temp.transpose(1,0,2))
+#            diag[s_aba:f_aba] += -temp.reshape(-1)
+#
+#            eris_OvOv_p = np.ascontiguousarray(eris_OOvv.transpose(0,2,1,3))
+#            eris_OvOv_p = eris_OvOv_p.reshape(nocc_b*nvir_a, nocc_b*nvir_a)
+#  
+#            temp = np.zeros((nvir_b, nocc_b, nvir_a))
+#            temp[:] += np.diagonal(eris_OvOv_p).reshape(nocc_b,nvir_a)
+#            temp = np.ascontiguousarray(temp.transpose(1,2,0))
+#            diag[s_bab:f_bab] += -temp.reshape(-1)
+#
+#            eris_oVoV_p = np.ascontiguousarray(eris_ooVV.transpose(0,2,1,3))
+#            eris_oVoV_p = eris_oVoV_p.reshape(nocc_a*nvir_b, nocc_a*nvir_b)
+#
+#            temp = np.zeros((nvir_a, nocc_a, nvir_b))
+#            temp[:] += np.diagonal(eris_oVoV_p).reshape(nocc_a,nvir_b)
+#            temp = np.ascontiguousarray(temp.transpose(1,2,0))
+#            diag[s_aba:f_aba] += -temp.reshape(-1)
+#        else :
+#           raise Exception("Precond not available for out-of-core and density-fitted algo")
 
```

Comment:
Why is this block commented?

Reply:
This block includes the ADC(2)-x and ADC(3) terms into the preconditioner but based on our tests, that does not improve convergence, and sometimes can make things a little worse. So, we commented out the terms as of now.